packages/arcgis-rest-request/src/job.ts

```diff
@@ -0,0 +1,494 @@
+   * @returns An new instance of Job class with the returned job id from submitJob request and requestOptions;
+   */
+  static submitJob(requestOptions: ISubmitJobOptions) {
+    const { url, params, startMonitoring, pollingRate, authentication } =
+      requestOptions;
+    const baseUrl = cleanUrl(url.replace(/\/submitJob\/?/, ""));
+    const submitUrl = baseUrl + "/submitJob";
+    return request(submitUrl, { params, authentication }).then(
+      (response) =>
+        new Job({
+          url: baseUrl,
+          authentication: authentication,
+          id: response.jobId,
+          startMonitoring,
+          pollingRate
+        })
+    );
+  }
+
+  /**
+   * The base URL of the job.
+   */
+  readonly url: string;
+  /**
+   * The job id indicating the specific job.
+   */
+  readonly id: string;
+  /**
+   * Authentication manager or access token to use for all job requests. 
+   */
+  readonly authentication: IAuthenticationManager | string;
+
+  /**
+   * Emitter is a third-party package used for our custom event listeners.
+   */
+  emitter: any;
+
+  /**
+   * Private pollingRate that is set if the user changes the pollingRate.
+   */
+  private _pollingRate: number;
+  /**
+   * Private boolean that checks to see if the user enables startMonitoring. 
+   */
+  private didUserEnableMonitoring: any;
+  /**
+   * Private method that calls setInterval();
+   */
+  private setIntervalHandler: any;
+
+  constructor(options: IJobOptions) {
+    const { url, id, pollingRate }: Partial<IJobOptions> = {
+      ...{ pollingRate: 5000, startMonitoring: true },
+      ...options
+    };
+
+    this.url = url; //user passes in the initial endpoint
+    this.id = id; //saved from the response from the static create job
+    this._pollingRate = pollingRate;
+    this.emitter = mitt(); //interval between each polling request
+    this.authentication = options.authentication;
+
+    /**
+     * If the startMonitoring from the requestOptions is set to true, {@linkcode Job.executePoll} will be called inside the startEventMonitoring.
+     */
+    if (options.startMonitoring) {
+      this.startEventMonitoring(pollingRate);
+    }
+  }
+
+  /**
+   * Getter that appends the job id to the base url.
+   */
+  private get jobUrl() {
+    return this.url + `/jobs/${this.id}`;
+  }
+
+  /**
+   * Getter that will return the boolean of setIntervalHandler.
+   */
+  get isMonitoring() {
+    return !!this.setIntervalHandler;
+  }
+
+  /**
+   * Getter that returns the private polling rate.
+   */
+  get pollingRate() {
+    return this._pollingRate;
+  }
+
+  /**
+   * Setter that sets a new polling rate if the user changes the rate during the job task.
+   */
+  set pollingRate(newRate: number) {
+    this.stopEventMonitoring();
+    this.startEventMonitoring(newRate);
+  }
+
+  /**
+   * Retrieves the status of the current job. 
+   * 
+   * @returns An object with the job id and jobStatus.
+   */
+  getJobInfo() {
```

Comment:
@marcysilverman it would be nice to build an interface (`IJobInfo`) that represents the response of the job info status call that includes all its properties.

```suggestion
  getJobInfo() : Promise<IJobInfo> {
```